renderer/src/components/Totals/Totals.tsx

```diff
@@ -123,9 +140,44 @@ const Totals = ({ selectedDate }) => {
     }
   };
 
+  const changePeriod = (periodName, selectedDate) => {
+    let dateName = "";
+
+    switch (periodName) {
+      case "day":
+        const today = new Date();
```

Comment:
May I ask you to create a separate function for the code from the 148 line to the 163 line?